### PR TITLE
Move utils to their own package, fixes #66.

### LINKED ISSUE #66: appveyor windows compile
I noticed something when trying to use the git repo directly in Cargo.toml.

```
rust-embed = { git = "https://github.com/pyros2097/rust-embed" }
```

Compiling with Appveyor, the build errored out with this message:

```
Compiling rust-embed-impl v4.4.0 (https://github.com/pyros2097/rust-embed#de8e97fe)
error: expected item, found `..`
 --> C:\Users\appveyor\.cargo\git\checkouts\rust-embed-88640eaf411660a2\de8e97f\impl\src\utils.rs:1:1
  |
1 | ../../src/utils.rs
  | ^^ expected item
error: aborting due to previous error
error: Could not compile `rust-embed-impl`.
```

The latest released version (v4.4.0) was building successfully.

### PR DESCRIPTION
**Pull request by AletiCodes**
This is my first PR in Rust so let me know if I done goofed, thanks!

### CODE REVIEW

File: utils/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+[package]
+name = "rust-embed-utils"
+version = "4.5.0"
+description = "Rust Custom Derive Macro which loads files into the rust binary at compile time during release and loads the file from the fs during dev"
```
Comment (Mcat12) [inline utils/Cargo.toml]: This description should reflect that this is simply a utils crate for rust-embed.

File: Cargo.toml [hunk 2]
```diff
@@ -13,6 +13,7 @@ authors = ["pyros2097 <pyros2097@gmail.com>"]
 [dependencies]
 walkdir = "2.2.7"
 rust-embed-impl = { version = "4.5.0", path = "impl"}
+rust-embed-utils = { version = "4.5.0", path = "utils"}
```
Comment (Mcat12) [inline Cargo.toml]: The `debug-embed` feature should also activate the utils' `debug-embed` feature (add it to the list below).

File: utils/readme.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# Rust Embed Implementation
```
Comment (Mcat12) [inline utils/readme.md]: Instead of "Implementation", use "Utilities"

File: utils/src/lib.rs [hunk 4]
```diff
@@ -1,15 +1,14 @@
...
-
+#[cfg(all(debug_assertions, not(feature = "debug-embed")))]
+extern crate walkdir;
 #[cfg_attr(all(debug_assertions, not(feature = "debug-embed")), allow(unused))]
 pub struct FileEntry {
   pub rel_path: String,
   pub full_canonical_path: String,
 }
 
+
```
Comment (Mcat12) [inline utils/src/lib.rs]: This extra space can be moved above `FileEntry`.

File: impl/src/lib.rs [hunk 5]
```diff
@@ -60,12 +59,13 @@ fn generate_assets(ident: &syn::Ident, folder_path: String) -> quote::Tokens {
 
 #[cfg(any(not(debug_assertions), feature = "debug-embed"))]
 fn generate_assets(ident: &syn::Ident, folder_path: String) -> quote::Tokens {
-  use utils::{get_files, FileEntry};
+  extern crate rust_embed_utils;
+  //use rust_embed_utils::{get_files, FileEntry};
```
Comment (Mcat12) [inline impl/src/lib.rs]: This commented out line can be removed.

**Comment by AletiCodes** [comment]
Thanks! I've made the changes based on your comments.

**Comment by pyros2097** [comment]
@AletiCodes Thanks for the PR. Awesome work. 
@Mcat12 Thanks for reviewing. LGTM.

**Pull request merged**